src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,34 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(actor);
+            if (session == null) {
+                throw new RuntimeException("There is no value to save!");
+            }
+            transaction.commit();
+        } catch (RuntimeException e) {
+            transaction.rollback();
+            throw new DataProcessingException("Error occurred during transaction", e);
```

Comment:
Why do you need to catch RuntimeException? I think you need to catch Exception. maybe you need to rewrite the structure, for example like this: catch (Exception e) {
            if (transaction != null) {
                transaction.rollback();
            }
            throw new DataProcessException
this is my opinion.